### PR TITLE
Verify identity quote before adding agent to the verifier

### PR DESCRIPTION
**Pull request by ansasaki**
This implements the enhancement proposed in https://github.com/keylime/enhancements/blob/master/66_quote_before_register.md

Swap the order of `do_cv()` and `do_quote()` calls in the tenant.

Fixes: #680

### CODE REVIEW

File: keylime/tenant.py [hunk 1]
```diff
@@ -1118,7 +1092,6 @@ def do_quote(self):
                 raise UserError(
                     f"Posting of Encrypted U to the Cloud Agent failed with response code {response.status_code} ({response.text})")
         except Exception as e:
```
Comment (THS-on) [inline keylime/tenant.py:1094]: The try-catch block can be fully removed, instead of capturing and re-raising the exception.

**Review by THS-on — CHANGES_REQUESTED** [review]
Can you check if the verification with `do_verify()` still works?

**Comment by kkaarreell** [comment]
Hi @ansasaki , I have found an issue with patch code coverage diff generating that manifests itself in your PR. I have committed a fix in https://github.com/RedHat-SP-Security/keylime-tests/pull/73. Would you mind rescheduling Packit tests in your PR (just F35 would be enough) so I can double-check it works fine? Thank you.

File: keylime/tenant.py (hunk 1 above)
Comment (ansasaki) [reply keylime/tenant.py:1094]: Thanks for the review, I'll remove it.

**Comment by ansasaki** [comment]
> Can you check if the verification with `do_verify()` still works?

I checked by passing `--verify` to the add command and it worked fine. However, when I tried with the rust agent I got errors on JSON parsing. 

> Hi @ansasaki , I have found an issue with patch code coverage diff generating that manifests itself in your PR. I have committed a fix in [RedHat-SP-Security/keylime-tests#73](https://github.com/RedHat-SP-Security/keylime-tests/pull/73). Would you mind rescheduling Packit tests in your PR (just F35 would be enough) so I can double-check it works fine? Thank you.

I pushed some changes which triggered re-runs for all tests.

**Comment by THS-on** [comment]
> I checked by passing --verify to the add command and it worked fine. However, when I tried with the rust agent I got errors on JSON parsing.

This seems like a bug in the rust agent.

**Comment by kkaarreell** [comment]
> I pushed some changes which triggered re-runs for all tests.

Right, I can see that the patch is correct now. Unfortunately, e2e tests do not utilize the modified code. Any suggestions for test scenarios (or patches) are welcome. :-)

**Comment by ansasaki** [comment]
> > I checked by passing --verify to the add command and it worked fine. However, when I tried with the rust agent I got errors on JSON parsing.
> 
> This seems like a bug in the rust agent.

It seems that when the rust agent returns a `400` error it does not include the `results` field. I need to debug it further.

**Comment by ansasaki** [comment]
> It seems that when the rust agent returns a `400` error it does not include the `results` field. I need to debug it further.

Indeed the responses for error cases were not wrapped in the JSON structure. I created a PR to fix this:

https://github.com/keylime/rust-keylime/pull/356

**Review by THS-on — APPROVED** [review]
LGTM.

**Pull request merged**